### DISCUSSION TITLE
Issues when building with flatpak

### DISCUSSION THREAD

**Discussion by ChildishGiant** [elementary/docs#103] [Q&A] [answered]
Has anyone had issues with flatpak? I'm going through the tutorial and got this when running flatpak-builder
```
Installing runtime/com.github.childishgiant.mixer.Locale/x86_64/master
Warning: Failed to install com.github.childishgiant.mixer.Locale: While pulling runtime/com.github.childishgiant.mixer.Locale/x86_64/master from remote mixer-origin: Content object ef2193a81c5fc70d8b43334e98f1f7e673b80d66e3200fadd9d179feee2d67a7: invalid mode 0100777 with bits 00002
Installing app/com.github.childishgiant.mixer/x86_64/master
Error: Failed to install com.github.childishgiant.mixer: While pulling app/com.github.childishgiant.mixer/x86_64/master from remote mixer-origin: Content object b86b10660863c62dbe28f1a05597efed3271b531c5d94f09932dc46584883cfe: invalid mode 0100777 with bits 00002
Install failed: Child process exited with code 1
```

**Comment by danirabbit** [comment]
I don't see a Flatpak manifest file on your GitHub repository: https://github.com/ChildishGiant/mixer. Can you push a branch with the manifest you're building? That would be helpful for finding the source of the issue you're experiencing

Also, just to confirm, is this output what you see when running `flatpak-builder build com.github.childishgiant.mixer.yml --user --install --force-clean` ?

**Reply by ChildishGiant** [reply]
I've just pushed my local changes, thank you for your help :smile: Yes, that's the exact command that I'm having issues with

**Comment by ChildishGiant** [comment]
![image](https://user-images.githubusercontent.com/13716824/125148976-fe3b2900-e12d-11eb-99a5-6fee8ece2a47.png)
Seems that another person has run into this issue

**Comment by ChildishGiant** [comment]
I'm still having issues with this 😔 Here's a full log of an attempted build [log.txt](https://github.com/elementary/docs/files/6844129/log.txt)

**Comment by ChildishGiant** [comment]
I've managed to get a build together by using `flatpak build-bundle` but this doesn't seem to work.
![image](https://user-images.githubusercontent.com/13716824/126345511-11d51d52-0494-49a2-a317-9f5daa88922f.png)

**Reply by danirabbit** [reply]
Hey sorry for the long response time! We're working on getting a new release of our Flatpak platform out this week and then I should be able to help app developers with packaging issues

Just wanted to make sure you know I'm seeing your posts just not able to dig into this just yet, sorry!

**Reply by ChildishGiant** [reply]
I've managed to fix this but have now run into yet another issue. My app relies on running command-line tools that listen to the pulseaudio daemon but because of the sandboxing it can't see that there's even one running. No rush to assist though, I know you have much bigger fish to fry right now :smile:

**Reply by danirabbit** [reply]
Instead of using the commandline, you'll want to use [`libpulse`](https://valadoc.org/libpulse/PulseAudio.html) and make sure that you've added `--socket=pulseaudio` to your flatpak manifest under `finish_args`

**Reply by ChildishGiant** [reply]
I've got it fully working with flatpak and do hope to convert to libpulse. I built it using pactl because it's a lot simpler to understand, as I'm learning vala while working on the project. I tried for a few hours to get libpulse working and failed so decided to move forward with the command line. Some dev docs on building bundles would be wonderful for easing the teething problems.

**Reply by Murmele** [reply]
> I've managed to fix this but have now run into yet another issue. My app relies on running command-line tools that listen to the pulseaudio daemon but because of the sandboxing it can't see that there's even one running. No rush to assist though, I know you have much bigger fish to fry right now smile

How did you fix this?

**Reply by ChildishGiant** [reply]
I still run into this fairly often, sometimes `sudo flatpak repair` or `flatpak repair --user` fixes it but other times I often just rely on github actions to build the flatpaks for me because it seems to work there :upside_down_face:

**Comment by ChildishGiant** [comment, marked as answer]
Turns out this is an issue with running flatpak-builder on a project on an NTFS drive apparently 🤷🏻‍♀️ https://github.com/flatpak/flatpak-builder/issues/462
